gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonClientCalls.java

```diff
@@ -115,21 +115,23 @@ public boolean setException(Throwable throwable) {
 
   private static class FutureListener<T> extends HttpJsonClientCall.Listener<T> {
     private final HttpJsonFuture<T> future;
+    private T storedMessage;
 
     private FutureListener(HttpJsonFuture<T> future) {
       this.future = future;
     }
 
     @Override
     public void onMessage(T message) {
-      if (!future.set(message)) {
+      if (storedMessage != null) {
         throw new IllegalStateException("More than one value received for unary call");
       }
+      storedMessage = message;
```

Comment:
Confirm: Is the future done on here? What if the runnableResult has an exception?